src/WithdrawKit.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity =0.8.17;
+
+import {PublicVault} from "src/PublicVault.sol";
+import {WithdrawProxy} from "src/WithdrawProxy.sol";
+import {ERC20} from "solmate/tokens/ERC20.sol";
+
+contract WithdrawKit {
+  error MinAmountError();
+  error LiensOpenForEpoch(uint256 epoch);
+  error FinalAuctionNotEnded(uint256 finalAuctionEnd);
+
+  function redeem(WithdrawProxy withdrawProxy, uint256 minAmountOut) external {
+    PublicVault publicVault = PublicVault(address(withdrawProxy.VAULT()));
+
+    //    ERC20 asset = publicVault.asset();
+
+    uint256 timedEpoch = getVaultEpochByBlockTime(publicVault);
+
+    uint256 epochDelta = withdrawProxy.CLAIMABLE_EPOCH() -
+      publicVault.getCurrentEpoch();
+    //epoch delta is the length of the epochdata array
+    (uint256 currentEpoch, , , , uint256 withdrawReserve, , ) = publicVault
+      .getPublicVaultState();
+
+    for (uint64 j = 0; j < epochDelta; j++) {
+      uint64 targetEpoch = publicVault.getCurrentEpoch() + j;
+      (uint256 liensOpenForEpoch, ) = publicVault.getEpochData(targetEpoch);
```

Comment:
If we process the epoch this will break the loop control because at the bottom of the for loop we called `processEpoch`